File: CSA/lab3/f32a/f32a.py
def sum_even_n(n):
    """Calculate the sum of even numbers from 1 to n"""
    if n <= 0:
        return -1
    total = 0
    for i in range(1, n + 1):
        if i % 2 == 0:
            total += i
    return total

print(sum_even_n(100000))
print(sum_even_n(110000))
print((2 ** 31 - 1) - sum_even_n(92680))
print((2 ** 31 - 1) - sum_even_n(92681))
print((2 ** 31 - 1) - sum_even_n(92682))

assert sum_even_n(5) == 6
assert sum_even_n(10) == 30
assert sum_even_n(90000) == 2025045000

File: CSA/lab3/f32a/f32a.c

#include <stdio.h>

int s(int n) {
    if (n <= 0)
        return -1;

    int total = 0;
    for (int i = 1; i < n + 1; ++i) {
        if (i % 2 == 0)
            total += i;
    }
    return total;
}

int main(void) {
    int n = 100000;

    printf("%d\n", s(n));
    printf("%d\n", (n / 2) * ((n / 2) + 1));
    
    return 0;
}
